tests/back/dataloaders/test_dataloaders.py

```diff
@@ -34,65 +35,57 @@ def test_json_dataloader_to_dataset():
+    json_binary = io.BytesIO(bytes(json_data, encoding="utf8"))
+    file = UploadFile(json_binary)
     with pytest.raises(ValueError):
-        test_dataset_path = "tests/back/dataloaders/irisDataset.json"
-        dataloader_test = JSONDataLoader()
-        params = {"data_ke": "data"}
-        with open(test_dataset_path, "r") as file:
-            json_data = file.read()
-        json_binary = io.BytesIO(bytes(json_data, encoding="utf8"))
-        file = UploadFile(json_binary)
         dataloader_test.load_data("tests/back/dataloaders", params, file=file)
-    assert True
 
 
 def test_wrong_path_create_csv_dataloader():
+    test_dataset_path = "tests/back/dataloaders/iris_unexisted.csv"
+    dataloader_test = CSVDataLoader()  # noqa: F841
+    params = {"separator": ","}  # noqa: F841
     with pytest.raises(FileNotFoundError):
-        test_dataset_path = "tests/back/dataloaders/iris_unexisted.csv"
-        dataloader_test = CSVDataLoader()
-        params = {"separator": ","}
         with open(test_dataset_path, "r") as file:
-            csv_data = file.read()
-        csv_binary = io.BytesIO(bytes(csv_data, encoding="utf8"))
-        file = UploadFile(csv_binary)
-        dataloader_test.load_data("tests/back/dataloaders", params, file=file)
-    assert True
+            csv_data = file.read()  # noqa: F841
 
 
 def test_wrong_path_create_json_dataloader():
+    test_dataset_path = "tests/back/dataloaders/irisDatasetUnexisted.json"
+    dataloader_test = JSONDataLoader()  # noqa: F841
+    params = {"data_key": "data"}  # noqa: F841
     with pytest.raises(FileNotFoundError):
-        test_dataset_path = "tests/back/dataloaders/irisDatasetUnexisted.json"
-        dataloader_test = JSONDataLoader()
-        params = {"data_key": "data"}
         with open(test_dataset_path, "r") as file:
-            json_data = file.read()
-        json_binary = io.BytesIO(bytes(json_data, encoding="utf8"))
-        file = UploadFile(json_binary)
-        dataloader_test.load_data("tests/back/dataloaders", params, file=file)
-    assert True
+            json_data = file.read()  # noqa: F841
 
 
 def test_invalidate_csv_dataloader():
+    test_dataset_path = "tests/back/dataloaders/wrong_iris.csv"
+    dataloader_test = CSVDataLoader()
+    params = {"separator": ","}
+    with open(test_dataset_path, "r") as file:
+        csv_data = file.read()
+    csv_binary = io.BytesIO(bytes(csv_data, encoding="utf8"))
+    file = UploadFile(csv_binary)
     with pytest.raises(DatasetGenerationError):
-        test_dataset_path = "tests/back/dataloaders/wrong_iris.csv"
-        dataloader_test = CSVDataLoader()
-        params = {"separator": ","}
-        with open(test_dataset_path, "r") as file:
-            csv_data = file.read()
-        csv_binary = io.BytesIO(bytes(csv_data, encoding="utf8"))
-        file = UploadFile(csv_binary)
         dataloader_test.load_data("tests/back/dataloaders", params, file=file)
-    assert True
+    os.remove(
```

Comment:
I think this remove should be inside the load_data function, in a finally statement